API.md

```diff
@@ -0,0 +1,106 @@
+## Models
+
+### Blueprint
+TBD
+
+### Header
+
+```
+{
+	type:      'header';
+	id:        number;
+	sourceUrl: string; // URL of the original page from which the header was created.
+	// More fields TBD.
+}
+```
+### Footer
+
+```
+{
+	type:      'footer';
+	id:        number;  
+	sourceUrl: string; // URL of the original page from which the footer was created.
+	// More fields TBD.
+}
+```
+### BlogPost
+
+```
+{
+	type:          'blog-post';
+	id:            number;  
+	transformedId: number;      // Id of the corresponding transformed post
+	sourceUrl:     string;      // URL of the source page from which this blog post was created.
+	date: {
+		type:     'date';
+	    original: string; // original HTML
+	    parsed:   string; // ISO date string
+	},
+	title: {
+		type:     'text';
+	    original: string; // original HTML
+	    parsed:   string; // plain text
+	},
+	content: {
+		type:     'html';
+	    original: string; // original HTML
+	    parsed:   string; // block markup
+	}
+}
+```
+
+## Subjects API
+
+These would work the same for any kind of subject, e.g. `blog-post`, `header`, or `footer`.
+
+### `create`
+
+**Request body**:
+
+```
+{
+	type: 'header' | 'footer' | 'blog-post' | ...;
```

Comment:
I am unsure of how things like header/footer/nav type would have a `sourceUrl` to begin with, since they may or may not change with different pages. Mostly won't change (should be our first consideration), but can change (should be our second consideration).

Reply:
Also, `type` gets used in the endpoint url, so the minimal request body just looks like:

```
{
	sourceUrl: string;
}
```

Reply:
I think we should just acknowledge that some subject types won't have a source url.

Reply:
To me, the fact that sourceUrl does not apply to header/footer/navigation is a sign that we should think of those in a different way than we do other subjects.

In reality, they are different, because there's only one (or a few) header/footer in a site, they are "global", or in other words part of the "layout" (aka theme) of the site, they aren't data in the same sense blog posts or products are.

My suggestion would be, as mentioned in another comment, to not tackle header/footer for now, and make it so that all subjects have a source url, because all "data subjects" need a source url.

Reply:
Agree, but then how do we handle navmenu?